### PR TITLE
applications: serial_lte_modem: Coverity Fix resource leak

### PR DESCRIPTION
**Pull request by junqingzou**
Close file descriptor (socket handle) in abnormal context.

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

### CODE REVIEW

**Review by lats1980 — COMMENTED** [review]
Since server_connect does not return error no after this change, line 380 becomes redundant.

			httpc.fd = INVALID_SOCKET;

**Comment by junqingzou** [comment]
> Since server_connect does not return error no after this change, line 380 becomes redundant.
> 
> ```
> 		httpc.fd = INVALID_SOCKET;
> ```

Updated

File: applications/serial_lte_modem/src/slm_at_tcpip.c [hunk 1]
```diff
@@ -466,25 +466,26 @@ static int do_listen(void)
...
+	int fd;
 	char peer_addr[INET_ADDRSTRLEN];
 	socklen_t len = sizeof(struct sockaddr_in);
 
-	ret = accept(client.sock, (struct sockaddr *)&remote, &len);
-	if (ret < 0) {
-		LOG_ERR("accept() failed: %d/%d", -errno, ret);
+	fd = accept(client.sock, (struct sockaddr *)&remote, &len);
+	if (fd == -1) {
+		LOG_ERR("accept() failed: %d", -errno);
```
Comment (lats1980) [inline applications/serial_lte_modem/src/slm_at_tcpip.c:475]: How about use INVALID_SOCKET instead of -1

**Review by lats1980 — APPROVED** [review]
Looks good to me.

File: applications/serial_lte_modem/src/slm_at_tcpip.c (hunk 1 above)
Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_at_tcpip.c:475]: Prefer to keep current code.
-1 is official BSD accept() return when API fails.
INVALID_SOCKET is SLM definition and happens to be -1.

**Review by rlubos — APPROVED** [review]

**Pull request merged**